### PR TITLE
Align assemblies autothreading

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._
made total-threads and in-parallel optional parameters for align-assemblies

```release-note
made total-threads and in-parallel optional parameters for align-assemblies
```

### CODE REVIEW

**Review by zrm22 — APPROVED** [review]

**Review by tcasstevens — APPROVED** [review]

File: docs/build_and_load.md [hunk 1]
```diff
@@ -407,7 +418,52 @@ or review the following code blocks:
 > prior `-R` and `-Q` parameters.
 
 
-[//]: # (#### Details - threads and parallelization)
+# (#### Details - threads and parallelization)
```
Comment (btmonier) [inline docs/build_and_load.md:421]: Change to:

```
#### Details - threads and parallelization
```

File: docs/build_and_load.md [hunk 2]
```diff
@@ -407,7 +418,52 @@ or review the following code blocks:
...
+Aligning with anchorwave is memory intensive and can be slow.  Processing speed may be increased by
+using multiple threads for each alignment, and/or by running multiple alignments in parallel.  The amount of memory
+each thread takes is dependent on the processor type.  The table below shows the memory usage for a single
+alignment based on processor type:
+
+| Processor | peak memory (Gb) | wall time |
```
Comment (btmonier) [inline docs/build_and_load.md:427]: Change to `GB`

File: docs/build_and_load.md [hunk 3]
```diff
@@ -407,7 +418,52 @@ or review the following code blocks:
...
+The `--total-threads` parameter indicates the total number of threads available for system use.  The `--in-parallel`
+parameter controls the number of alignments run in parallel.  When these values are not specified, the software will
+compute the optimal values based on the system processor and memory configuration. 
+
+The number of threads that may be run in parallel is limited by the amount of memory available.  The
+system is queried for memory and processor information.  The number of threads that may be run in parallel
+is determined by "system memory" / "peak memory" from the table above.  To generalize the calculation, we divide
+memory available (Gib) by 21 and round down to the nearest integer.
+
+For example, if the system has 512 Gb of memory, 80 processors and 5 assemblies that need aligning,
```
Comment (btmonier) [inline docs/build_and_load.md:444]: Change to `GB`

File: docs/build_and_load.md [hunk 4]
```diff
@@ -407,7 +418,52 @@ or review the following code blocks:
...
+| ARM       | 34.2             | 18:08:57  |
+
+The `--total-threads` parameter indicates the total number of threads available for system use.  The `--in-parallel`
+parameter controls the number of alignments run in parallel.  When these values are not specified, the software will
+compute the optimal values based on the system processor and memory configuration. 
+
+The number of threads that may be run in parallel is limited by the amount of memory available.  The
+system is queried for memory and processor information.  The number of threads that may be run in parallel
+is determined by "system memory" / "peak memory" from the table above.  To generalize the calculation, we divide
+memory available (Gib) by 21 and round down to the nearest integer.
```
Comment (btmonier) [inline docs/build_and_load.md:442]: Change to `GiB`

File: docs/build_and_load.md [hunk 5]
```diff
@@ -442,6 +498,38 @@ or review the following code blocks:
 
 [//]: # (| ARM       | 34.2             | 18:08:57  |)
 
+### Annotate Fastas
```
Comment (btmonier) [inline docs/build_and_load.md:501]: This section is actually written out in further detail under the "Compress FASTA files" section.

File: docs/build_and_load.md (hunk 1 above)
Comment (lynnjo) [reply docs/build_and_load.md:421]: done

File: docs/build_and_load.md (hunk 2 above)
Comment (lynnjo) [reply docs/build_and_load.md:427]: done

File: docs/build_and_load.md (hunk 3 above)
Comment (lynnjo) [reply docs/build_and_load.md:444]: done

File: docs/build_and_load.md (hunk 4 above)
Comment (lynnjo) [reply docs/build_and_load.md:442]: done

File: docs/build_and_load.md (hunk 5 above)
Comment (lynnjo) [reply docs/build_and_load.md:501]: You are changing this, correct?

Comment (lynnjo) [reply docs/build_and_load.md:501]: Ie, deleting what I have, moving your stuff up where it makes sense? Or I can add it - let me know.

Comment (btmonier) [reply docs/build_and_load.md:501]: This has been moved and corrected collaboratively, so I will mark this as resolved

**Review by btmonier — APPROVED** [review]

**Pull request merged**